lib/monitoring/aws-rds/RdsClusterMetricFactory.ts

```diff
@@ -14,11 +18,12 @@ export interface RdsClusterMetricFactoryProps {
   /**
    * database cluster (either this or `clusterIdentifier` need to be specified)
    */
-  readonly cluster?: IDatabaseCluster;
+  readonly cluster?: IDatabaseCluster | ServerlessCluster;
 }
 
 export class RdsClusterMetricFactory {
   readonly clusterIdentifier: string;
+  readonly cluster?: IDatabaseCluster | ServerlessCluster;
```

Comment:
You could make this `DatabaseCluster | ServerlessCluster`.

For the type guard, you'd need to basically have something that can differentiate between the two:
- https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseCluster.html
- https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.ServerlessCluster.html

For example:

```typescript
function isServerlessCluster(obj: any): obj is ServerlessCluster {
  return (obj as any).clusterArn !== undefined;
}
```

This ought to work since `clusterArn` is unique to `ServerlessCluster ` amongst the two types.

Reply:
I'll try that out. Thanks!